monkey/monkey_island/cc/server_setup.py

```diff
@@ -13,41 +15,71 @@
     sys.path.insert(0, MONKEY_ISLAND_DIR_BASE_PATH)
 
 import monkey_island.cc.environment.environment_singleton as env_singleton  # noqa: E402
+import monkey_island.setup.config_setup as config_setup  # noqa: E402
 from common.version import get_version  # noqa: E402
 from monkey_island.cc.app import init_app  # noqa: E402
+from monkey_island.cc.arg_parser import IslandCmdArgs  # noqa: E402
+from monkey_island.cc.arg_parser import parse_cli_args  # noqa: E402
 from monkey_island.cc.resources.monkey_download import MonkeyDownload  # noqa: E402
 from monkey_island.cc.server_utils.bootloader_server import BootloaderHttpServer  # noqa: E402
 from monkey_island.cc.server_utils.consts import MONKEY_ISLAND_ABS_PATH  # noqa: E402
 from monkey_island.cc.server_utils.encryptor import initialize_encryptor  # noqa: E402
+from monkey_island.cc.server_utils.island_logger import reset_logger, setup_logging  # noqa: E402
 from monkey_island.cc.services.initialize import initialize_services  # noqa: E402
 from monkey_island.cc.services.reporting.exporter_init import populate_exporter_list  # noqa: E402
 from monkey_island.cc.services.utils.network_utils import local_ip_addresses  # noqa: E402
 from monkey_island.cc.setup.mongo.database_initializer import init_collections  # noqa: E402
-from monkey_island.cc.setup.mongo.mongo_setup import MONGO_URL, setup_mongodb  # noqa: E402
+from monkey_island.cc.setup.mongo.mongo_setup import (  # noqa: E402
+    MONGO_URL,
+    connect_to_mongodb,
+    register_mongo_shutdown_callback,
+    start_mongodb,
+)
 from monkey_island.setup.island_config_options import IslandConfigOptions  # noqa: E402
 
 logger = logging.getLogger(__name__)
 
 
-def setup_island(setup_only: bool, config_options: IslandConfigOptions, server_config_path: str):
-    env_singleton.initialize_from_file(server_config_path)
+def run_monkey_island():
+    island_args = parse_cli_args()
+    config_options, server_config_path = _setup_data_dir(island_args)
 
-    initialize_encryptor(config_options.data_dir)
-    initialize_services(config_options.data_dir)
+    _configure_logging(config_options)
+    _initialize_global_resources(config_options, server_config_path)
 
-    bootloader_server_thread = Thread(
-        target=BootloaderHttpServer(MONGO_URL).serve_forever, daemon=True
-    )
+    if config_options.start_mongodb:
+        mongo_db_process = start_mongodb(config_options.data_dir)
+        register_mongo_shutdown_callback(mongo_db_process)
 
-    bootloader_server_thread.start()
-    _start_island_server(setup_only, config_options)
-    bootloader_server_thread.join()
+    connect_to_mongodb()
 
+    _start_island_server(island_args.setup_only, config_options)
 
-def _start_island_server(should_setup_only, config_options: IslandConfigOptions):
 
-    setup_mongodb(config_options)
+def _setup_data_dir(island_args: IslandCmdArgs) -> Tuple[IslandConfigOptions, str]:
```

Comment:
The name `setup_data_dir` implies that the data directory will be created if needed and proper permissions will be set for it. Now it returns island configuration arguments and island configuration path, which makes no sense. I think this should be 3 separate methods:
1. `get_island_config_options(island_args) -> IslandConfigOptions`
2. `setup_data_dir(island_config_options) -> data_dir_path`
3. `setup_server_config_json_file(IslandConfigOptions, data_dir_path)`

Reply:
Not in scope for this PR, but we should make a decision about it and either rename the method back to a vague name it had, or create a PR right after this one to fix.

Reply:
Using a vague name that doesn't communicate what the function does is never a good solution. I'll break it up.

Reply:
@VakarisZ I've tried to refactor this, but I've yet to come up with a solution that is significantly better. The real issue is that we need to distinguish whether or not the user provided a server_config.json or if they're using the default.

The solution to this is https://github.com/guardicore/monkey/issues/1206. Once we fix #1206, this code will be much simpler to clean up (half of it will be gone).